src/rubyfmt.rb

```diff
@@ -716,6 +716,25 @@ def format_int(ps, rest)
   ps.emit_indent if ps.start_of_line.last
 
   int = rest[0]
+
+  if int[0] != "0"
+    int_array = []
```

Comment:
Rather than keeping an array and a counter, could the each block be expressed as a `reduce` instead?

Reply:
Currently, the counter is needed to keep track of how many digits we've placed since a valid Ruby integer of length N can contain up to N-1 underscores (e.g. 123 vs 1_2_3), though if we combine it with the rejection of underscores as proposed above it's certainly possible. However, then we need to determine the placement of the underscore based on the length of the string, which might make the logic a little less clear?